solana/program_v2/programs/gateway_v2/src/state/gatekeeper.rs

```diff
@@ -43,6 +43,10 @@ impl Gatekeeper {
             + OC_SIZE_VEC_PREFIX + GatekeeperAuthKey::ON_CHAIN_SIZE * auth_keys
         // auth keys
     }
+    // Checks if the network_token_account is a valid token account for the network
+    pub fn check_network_ownership(&self, gatekeeper: &[Pubkey], owner: Pubkey) -> bool {
+        gatekeeper.iter().any(|&key| key == owner)
```

Comment:
I'm not sure I understand this. The network_token_account is owned by the network and has no relationship to the Gatekeepers.

Reply:
I think what you want to check is if `gatekeeper_token_account` belongs to any GATEKEEPER in the network? (is this sufficient)? Don't you KNOW the actual gatekeeper during issue?

Reply:
So I would not expect that you iterate.